pkg/controllers/plugin/pluginpreset_controller_test.go

```diff
@@ -242,6 +242,260 @@ var _ = Describe("PluginPreset Controller Lifecycle", Ordered, func() {
+				Name: pluginPresetName,
+			},
+			Spec: greenhousev1alpha1.PluginPresetSpec{
+				Plugin: greenhousev1alpha1.PluginSpec{
+					PluginDefinition: pluginPresetDefinitionName,
+					OptionValues: []greenhousev1alpha1.PluginOptionValue{
+						{
+							Name:  "plugin_preset.test_parameter",
+							Value: asAPIextensionJSON(3),
+						},
+					},
+				},
+			},
+		}
+
+		Expect(shouldSkipPlugin(testPlugin, testPresetCase)).To(BeFalse())
+	})
+
+	It("should skip when plugin preset has the same values like plugin", func() {
+		testPlugin := &greenhousev1alpha1.Plugin{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					greenhouseapis.LabelKeyPluginPreset: pluginPresetName,
+				},
+			},
+			Spec: greenhousev1alpha1.PluginSpec{
+				PluginDefinition: pluginPresetDefinitionName,
+				OptionValues: []greenhousev1alpha1.PluginOptionValue{
+					{
+						Name:  "global.greenhouse.test_parameter",
+						Value: asAPIextensionJSON(2),
+					},
+					{
+						Name:  "plugin_preset.test_parameter",
+						Value: asAPIextensionJSON(3),
+					},
+				},
+			},
+		}
+
+		testPresetCase := &greenhousev1alpha1.PluginPreset{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pluginPresetName,
+			},
+			Spec: greenhousev1alpha1.PluginPresetSpec{
+				Plugin: greenhousev1alpha1.PluginSpec{
+					PluginDefinition: pluginPresetDefinitionName,
+					OptionValues: []greenhousev1alpha1.PluginOptionValue{
+						{
+							Name:  "plugin_preset.test_parameter",
+							Value: asAPIextensionJSON(3),
+						},
+					},
+				},
+			},
+		}
+
+		Expect(shouldSkipPlugin(testPlugin, testPresetCase)).To(BeTrue())
+	})
+
+	It("should not skip when plugin has custom values", func() {
+		testPlugin := &greenhousev1alpha1.Plugin{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					greenhouseapis.LabelKeyPluginPreset: pluginPresetName,
+				},
+			},
+			Spec: greenhousev1alpha1.PluginSpec{
+				PluginDefinition: pluginPresetDefinitionName,
+				OptionValues: []greenhousev1alpha1.PluginOptionValue{
+					{
+						Name:  "greenhouse.global.test_parameter",
+						Value: asAPIextensionJSON(2),
+					},
+					{
+						Name:  "plugin_preset.test_parameter",
+						Value: asAPIextensionJSON(3),
+					},
+					{
+						Name:  "custom_parameter",
+						Value: asAPIextensionJSON(123),
+					},
+				},
+			},
+		}
+
+		testPresetCase := &greenhousev1alpha1.PluginPreset{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pluginPresetName,
+			},
+			Spec: greenhousev1alpha1.PluginPresetSpec{
+				Plugin: greenhousev1alpha1.PluginSpec{
+					PluginDefinition: pluginPresetDefinitionName,
+					OptionValues: []greenhousev1alpha1.PluginOptionValue{
+						{
+							Name:  "plugin_preset.test_parameter",
+							Value: asAPIextensionJSON(3),
+						},
+					},
+				},
+			},
+		}
+
+		Expect(shouldSkipPlugin(testPlugin, testPresetCase)).To(BeFalse())
+	})
+
+	It("should skip when plugin preset has the same values like plugin", func() {
```

Comment:
Let's add a test for defaulted values as mentioned here... :)

Reply:
I added the test for defaulted values in PluginPreset lifecycle test.